### PR TITLE
fix: Meetings: improve top bar layout on mobile

### PR DESCRIPTION
**Pull request by gitstart**
# Description

Fixes/Partially Fixes [issue](https://github.com/ParabolInc/parabol/issues/7143)
Issue #7143

- Improves the standup meeting topbar user experience when the meeting title is too long on mobile view.
- Helps to notify user when the max number of characters allowable has been reached for mobile and normal view.
- Improves general look of the standup topbar on mobile view.

## Demo
https://user-images.githubusercontent.com/116144318/221913817-34a7b1f2-d32f-4a94-81ba-5f982e8bb303.mov

## Testing scenarios

[Please list all the testing scenarios a reviewer has to check before approving the PR]

- [x] Improve mobile view layout for Standup Topbar.   
- [x] Notify the user when the standup meeting name limit has been reached.
- [x] Notify the user when the organization name limit has been reached.

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

**Comment by gitstart-parabol** [comment]
@BartoszJarocki this PR is ready for review

File: packages/client/components/BackButton.tsx [hunk 1]
```diff
@@ -17,6 +17,8 @@ const IconButton = styled(FlatButton)({
 
 const BackIcon = styled(ArrowBack)({
   color: 'inherit'
+  position: 'fixed',
+  bottom: '94%'
```
Comment (BartoszJarocki) [inline packages/client/components/BackButton.tsx:21]: This doesn't look good, why hardcode something that looks like a very arbitrary number?

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]

**Comment by gitstart-parabol** [comment]
Hi @BartoszJarocki, what's the verdict on this?

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]

File: packages/client/validation/Legitity.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ class Legitity {
     // this.value.length gives us the count of UTF-16 units, so 🔥 has a length of 2
     // Spreading the string into an array gives us the desired length in codepoints (characters): https://stackoverflow.com/a/54369605
     const value = [...this.value]
-    if (!this.error && value.length > len) {
+    if (!this.error && value.length >= len) {
```
[the remark this answers is not in the document]
Comment (gitstart-parabol) [reply packages/client/validation/Legitity.ts:42]: @BartoszJarocki the current code only checks if the length of the input is greater than the specified error case length, which is why the error indicator does not appear when the `max` for the standup 'title' is reached; however, this change will be improved.

**Comment by gitstart-parabol** [comment]
@BartoszJarocki this PR is ready for review.

**Review by BartoszJarocki — APPROVED** [review]
looks good! @Dschoordsch do you mind checking it on Android? thanks!

File: packages/client/validation/Legitity.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dschoordsch) [reply packages/client/validation/Legitity.ts]: -2 Changing this breaks the intend of the function. I expect `new Legitity('123').max(3)` to pass, not to fail. I made this explicit in #7863.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
See Legitity comment

File: packages/client/validation/Legitity.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gitstart-parabol) [reply packages/client/validation/Legitity.ts]: Okay, we'll revert this change to reinstate the desired behaviour 🙌

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
The warning does not appear anymore

**Comment by gitstart-parabol** [comment]
> The warning does not appear anymore

Hi @Dschoordsch, after a bit of investigation, we discovered that the existing validator `Legitity.max` will never catch any value greater than the maximum length because the `maxLength` is also being passed down to the `TextArea` and `Input` components. Because of this, the input field constraints prevent the user from entering any text that would trigger the warning.

Is there any other strategy you can suggest to get a more preferred outcome?

**Comment by Dschoordsch** [comment]
The behaviour we want is:
* the user can input characters up to maxlength without any warning
* if the user writes the character maxlength+1 we show the warning, that means we should remove maxlength from the text input
* we could either forbid saving or truncate the text to maxlength on saving. I would argue the latter is more convenient and thus we should go this way

**Comment by gitstart-parabol** [comment]
@Dschoordsch, This PR awaits another review.

File: packages/client/components/EditableText.tsx [hunk 3]
```diff
@@ -166,7 +166,7 @@ const EditableText = forwardRef((props: Props, ref: any) => {
     const inProps = {
       autoFocus: true,
       ref: inputRef as any,
-      maxLength,
+      maxLength: maxLength + 1,
```
Comment (Dschoordsch) [inline packages/client/components/EditableText.tsx:169]: -2 This component is used elsewhere, please don't change global components to fix a local issue without retesting all other usages of those components.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

File: packages/client/components/EditableText.tsx (hunk 3 above)
Comment (gitstart-parabol) [reply packages/client/components/EditableText.tsx:169]: Hello @Dschoordsch , regarding the general fix in this PR, we figured that the bug mentioned  in this issue seemed to be occurring in other areas where this component was being used as well. For instance, when updating the `organization name`.

Comment (Dschoordsch) [reply packages/client/components/EditableText.tsx:169]: Can you please update the testing scenarios to the new changes?

Comment (gitstart-parabol) [reply packages/client/components/EditableText.tsx:169]: Hi @Dschoordsch, we are unable to locate any ui tests that require updating on this instance. This recent change only result in a visual difference, without affecting the current unit test suite for the `Legality` file.

Comment (Dschoordsch) [reply packages/client/components/EditableText.tsx:169]: @gitstart-parabol sorry for not being more clear what I meant: Could you please update the manual test cases mentioned in the description of this PR to also test for regressions in other places where your changes apply?

**Comment by mattkrick** [comment]
Hi! Thank you for contributing to our app! We're closing this PR due to inactivity, but that doesn't mean the end of the code! This is still a useful feature & if we prioritize it in an upcoming sprint we will use this PR as a reference point.

**Pull request closed without merging**